[Discover] Runtime Fields Editor integration: Add + Delete operation

majagrubic opened the pull request:
Release Notes: Adds runtime field editor to Discover

## Summary

This is a continuation of integration of runtime fields editor in Discover, by adding Add and Delete operation, similar to how was done in Lens.

Adding a new field:
<img width="576" alt="Screenshot 2021-04-12 at 15 56 23" src="https://user-images.githubusercontent.com/1937956/114415654-cc0a4180-9ba7-11eb-9754-4f2e53e134fc.png">


Deleting a runtime field:
<img width="609" alt="Screenshot 2021-04-12 at 15 59 24" src="https://user-images.githubusercontent.com/1937956/114415970-17245480-9ba8-11eb-9bd9-6990a5c062dc.png">


Note: When there is an embeddable that depends on a runtime field, and the field is deleted, we show the field in the embeddable, with no values, but it can be removed from the embeddable itself. This UX is not great, but I think we agreed it's good enough for now.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the [cloud](https://github.com/elastic/cloud) and added to the [docker list](https://github.com/elastic/kibana/blob/c29adfef29e921cc447d2a5ed06ac2047ceab552/src/dev/build/tasks/os_packages/docker_generator/resources/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/)
* Commit: 6b1bb4c418892f051f999a89ebd27f76e098a1c5
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/execution/node/418/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/","number":"118631","commit":"6b1bb4c418892f051f999a89ebd27f76e098a1c5"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/)
* Commit: b4a3003ba8b05d5328152b3962b0b1940d71e67f
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96762/b4a3003ba8b05d5328152b3962b0b1940d71e67f)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/execution/node/483/log)

### History
* :broken_heart: [Build #118631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/) failed 6b1bb4c418892f051f999a89ebd27f76e098a1c5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/","number":"118631","commit":"6b1bb4c418892f051f999a89ebd27f76e098a1c5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/","number":"118703","commit":"b4a3003ba8b05d5328152b3962b0b1940d71e67f"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/)
* Commit: 60151274baa6ef0017a0802a4e33a5ed0688c679
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96762/60151274baa6ef0017a0802a4e33a5ed0688c679)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2ab94f05e1e8846c77a41cfc36eaa722b207f80e) | [after](https://github.com/elastic/kibana/commit/60151274baa6ef0017a0802a4e33a5ed0688c679) | diff |
| --- | --- | --- | --- |
| `discover` | 397.0KB | 401.2KB | +4.1KB |

### History
* :broken_heart: [Build #118703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/) failed b4a3003ba8b05d5328152b3962b0b1940d71e67f
* :broken_heart: [Build #118631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/) failed 6b1bb4c418892f051f999a89ebd27f76e098a1c5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/","number":"118631","commit":"6b1bb4c418892f051f999a89ebd27f76e098a1c5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/","number":"118703","commit":"b4a3003ba8b05d5328152b3962b0b1940d71e67f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/","number":"118753","commit":"60151274baa6ef0017a0802a4e33a5ed0688c679"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

majagrubic commented on src/plugins/discover/public/application/components/sidebar/discover_field.tsx:
```diff
@@ -311,6 +324,29 @@ export function DiscoverField({
+                data-test-subj={`discoverFieldListPanelDelete-${field.name}`}
+                color="danger"
+                aria-label={i18n.translate('discover.fieldChooser.discoverField.deleteFieldLabel', {
+                  defaultMessage: 'Delete index pattern field',
```
@flash1293 in Discover we refer to 'remove' as removing the field from the table, so I've decided to use `Delete` here. Think we could change this in Lens so well, so it's consistent?

flash1293 replied:
That makes a lot of sense, index pattern management is also using "delete" instead of "remove":
<img width="289" alt="Screenshot 2021-04-13 at 11 10 33" src="https://user-images.githubusercontent.com/1508364/114528072-f19b5780-9c48-11eb-8670-7c9898e744fc.png">

ryankeairns commented:
@MichaelMarcialis you'd have the best read on this for the Kibana Design review. Thanks!

flash1293 commented on src/plugins/discover/public/application/components/doc/use_es_doc_search.ts:
```diff
@@ -42,6 +42,7 @@ export function buildSearchBody(
     fields: useNewFieldsApi ? [{ field: '*', include_unmapped: 'true' }] : undefined,
     script_fields: computedFields.scriptFields,
     docvalue_fields: computedFields.docvalueFields,
+    runtime_mappings: useNewFieldsApi && runtimeFields ? runtimeFields : {},
```
Is this related to the add/delete change? It seems like this is an unrelated bug fix, right?

majagrubic replied:
No, this is needed for a single doc view, so that runtime index pattern fields are shown in a single doc view

flash1293 replied:
But this is not related to adding/deleting runtime fields from within Discover, right? if it's a bug which would be in there without the changes of this PR (and especially if it's in 7.12) I think it's worth at least a separate issue closed by this PR and a fix release note so we can point to something.

flash1293 commented on src/plugins/discover/public/application/components/sidebar/discover_field.tsx:
```diff
@@ -289,7 +301,8 @@ export function DiscoverField({
   const isRuntimeField = Boolean(indexPattern.getFieldByName(field.name)?.runtimeField);
   const isUnknownField = field.type === 'unknown' || field.type === 'unknown_selected';
   const canEditField = onEditField && (!isUnknownField || isRuntimeField);
-  const displayNameGrow = canEditField ? 9 : 10;
```
can you explain the grow stuff? It seems like this would be way simpler by just using `grow={true}` on the display name and `grow={false}` on the buttons (which is what Lens is doing: https://github.com/elastic/kibana/blob/c218ce292af1cc90ce619fbc4f1b7459e3255d4b/x-pack/plugins/lens/public/indexpattern_datasource/field_item.tsx#L334 )

majagrubic replied:
I spent almost a day trying to get this to align nicely 😅 It was ugly. Will try it once again

flash1293 commented on src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx:
```diff
@@ -224,6 +229,38 @@ export function DiscoverSidebar({
+        ? async (fieldName: string) => {
+            const ref = indexPatternFieldEditor.openDeleteModal({
+              ctx: {
+                // @ts-ignore
```
Nit: I'm always scared by `@ts-ignore` as they can shadow future legitimate errors. What about just extending the condition in line 234: `canEditIndexPatternField && selectedIndexPattern`. This will resolve the type error without reducing type safety (like for example if the index pattern type changes in the future or something)

flash1293 commented on src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx:
```diff
@@ -294,14 +393,19 @@ export function DiscoverSidebar({
-            state={state}
-            setAppState={setAppState}
-          />
+          <EuiFlexGroup direction="row" alignItems="center">
```
This is using medium gutter width which is too much here IMHO

Discover:
<img width="380" alt="Screenshot 2021-04-13 at 11 42 41" src="https://user-images.githubusercontent.com/1508364/114532698-5f498280-9c4d-11eb-9618-b59013bbf2a3.png">

Lens:
<img width="521" alt="Screenshot 2021-04-13 at 11 43 07" src="https://user-images.githubusercontent.com/1508364/114532754-71c3bc00-9c4d-11eb-8c2b-c856468d47d1.png">

flash1293 commented on src/plugins/discover/public/application/components/sidebar/discover_field.tsx:
```diff
@@ -311,6 +324,29 @@ export function DiscoverField({
           </EuiFlexItem>
         )}
+        {canDeleteField && (
+          <EuiFlexItem grow={1} data-test-subj="discoverFieldListPanelDeleteItem">
```
This flex group uses a medium gutter width which adds a lot of space between the buttons:
Discover:
<img width="314" alt="Screenshot 2021-04-13 at 11 44 33" src="https://user-images.githubusercontent.com/1508364/114532955-aafc2c00-9c4d-11eb-9d36-66c987837887.png">

Lens:
<img width="303" alt="Screenshot 2021-04-13 at 11 44 42" src="https://user-images.githubusercontent.com/1508364/114532964-acc5ef80-9c4d-11eb-9e44-219adb102a8a.png">

majagrubic replied:
Changed to `xs` gutter size

flash1293 reviewed (commented):
Looks mostly good to me, left some nits

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119168/)
* Commit: b928addc472d9e74ea855c7617dd8f0b6e5d177c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96762/b928addc472d9e74ea855c7617dd8f0b6e5d177c)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2ab94f05e1e8846c77a41cfc36eaa722b207f80e) | [after](https://github.com/elastic/kibana/commit/b928addc472d9e74ea855c7617dd8f0b6e5d177c) | diff |
| --- | --- | --- | --- |
| `discover` | 397.0KB | 401.2KB | +4.1KB |

### History
* :green_heart: [Build #118753](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/) succeeded 60151274baa6ef0017a0802a4e33a5ed0688c679
* :broken_heart: [Build #118703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/) failed b4a3003ba8b05d5328152b3962b0b1940d71e67f
* :broken_heart: [Build #118631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/) failed 6b1bb4c418892f051f999a89ebd27f76e098a1c5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/","number":"118631","commit":"6b1bb4c418892f051f999a89ebd27f76e098a1c5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/","number":"118703","commit":"b4a3003ba8b05d5328152b3962b0b1940d71e67f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/","number":"118753","commit":"60151274baa6ef0017a0802a4e33a5ed0688c679"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119168/","number":"119168","commit":"b928addc472d9e74ea855c7617dd8f0b6e5d177c"}]}
PIPELINE-->

MichaelMarcialis commented on src/plugins/discover/public/application/components/sidebar/discover_field.tsx:
```diff
@@ -289,13 +301,13 @@ export function DiscoverField({
-      <EuiFlexGroup responsive={false}>
-        <EuiFlexItem grow={displayNameGrow}>{field.displayName}</EuiFlexItem>
+      <EuiFlexGroup responsive={false} gutterSize="xs">
+        <EuiFlexItem grow={true}>{field.displayName}</EuiFlexItem>
```
Consider wrapping `{field.displayName}` in a heading element. For Lens, an `h5` made the most sense, but you may want a different level here, depending on your existing heading hierarchy.

MichaelMarcialis commented on src/plugins/discover/public/application/components/sidebar/discover_field.tsx:
```diff
@@ -289,13 +301,13 @@ export function DiscoverField({
     <EuiPopoverTitle style={{ textTransform: 'none' }} className="eui-textBreakWord">
-      <EuiFlexGroup responsive={false}>
-        <EuiFlexItem grow={displayNameGrow}>{field.displayName}</EuiFlexItem>
+      <EuiFlexGroup responsive={false} gutterSize="xs">
```
If there's a desire to match Lens, this can be changed to the following:

```suggestion
      <EuiFlexGroup responsive={false} gutterSize="s">
```

MichaelMarcialis reviewed (approved):
Left two small nitpick comments, but nothing worth holding you up for. Approving now.

So we changed direction and decided to include the delete option directly from the Discover and Lens field lists after all? Am I correct in assuming that both the save runtime field toast and delete runtime field modal components come from the team working on runtime fields? Just curious, as there may be a few opportunities for tweaks there, but I imagine they're not a part of this PR. Thanks!

majagrubic commented:
Thanks for the review @MichaelMarcialis. I've addressed your suggestions. Now to answer your questions:
1. Yes, we decided to go with deleting the field in both Lens & Discover
2. Toast and the runtime field editor are indeed owned by the es-ui team

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119511/)
* Commit: b0093d1c0c78e0831ca4e2628f0706e0a189dd83
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96762/b0093d1c0c78e0831ca4e2628f0706e0a189dd83)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2ab94f05e1e8846c77a41cfc36eaa722b207f80e) | [after](https://github.com/elastic/kibana/commit/b0093d1c0c78e0831ca4e2628f0706e0a189dd83) | diff |
| --- | --- | --- | --- |
| `discover` | 397.0KB | 401.2KB | +4.2KB |

### History
* :green_heart: [Build #119168](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119168/) succeeded b928addc472d9e74ea855c7617dd8f0b6e5d177c
* :green_heart: [Build #118753](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/) succeeded 60151274baa6ef0017a0802a4e33a5ed0688c679
* :broken_heart: [Build #118703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/) failed b4a3003ba8b05d5328152b3962b0b1940d71e67f
* :broken_heart: [Build #118631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/) failed 6b1bb4c418892f051f999a89ebd27f76e098a1c5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/","number":"118631","commit":"6b1bb4c418892f051f999a89ebd27f76e098a1c5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/","number":"118703","commit":"b4a3003ba8b05d5328152b3962b0b1940d71e67f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/","number":"118753","commit":"60151274baa6ef0017a0802a4e33a5ed0688c679"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119168/","number":"119168","commit":"b928addc472d9e74ea855c7617dd8f0b6e5d177c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119511/","number":"119511","commit":"b0093d1c0c78e0831ca4e2628f0706e0a189dd83"}]}
PIPELINE-->

majagrubic commented:
@elasticmachine merge upstream

flash1293 reviewed (approved):
LGTM, works fine. Great job!

kertal reviewed (commented):
sorry for being late at the party, I think I've found an issue concerting "mobile" view, seems it's not possible to add a field here:
![image](https://user-images.githubusercontent.com/463851/114746962-981e5000-9d50-11eb-988d-e726aecd5bba.gif)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119631/)
* Commit: 104717da112829974f98351b2e3394dc39da972e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96762/104717da112829974f98351b2e3394dc39da972e)
* [Documentation Changes](https://kibana_96762.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ad628878b11f1252769a511b24a7c76a3dbe046a) | [after](https://github.com/elastic/kibana/commit/104717da112829974f98351b2e3394dc39da972e) | diff |
| --- | --- | --- | --- |
| `discover` | 402.7KB | 406.9KB | +4.2KB |

### History
* :green_heart: [Build #119511](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119511/) succeeded b0093d1c0c78e0831ca4e2628f0706e0a189dd83
* :green_heart: [Build #119168](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119168/) succeeded b928addc472d9e74ea855c7617dd8f0b6e5d177c
* :green_heart: [Build #118753](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/) succeeded 60151274baa6ef0017a0802a4e33a5ed0688c679
* :broken_heart: [Build #118703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/) failed b4a3003ba8b05d5328152b3962b0b1940d71e67f
* :broken_heart: [Build #118631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/) failed 6b1bb4c418892f051f999a89ebd27f76e098a1c5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118631/","number":"118631","commit":"6b1bb4c418892f051f999a89ebd27f76e098a1c5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/","number":"118703","commit":"b4a3003ba8b05d5328152b3962b0b1940d71e67f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/","number":"118753","commit":"60151274baa6ef0017a0802a4e33a5ed0688c679"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119168/","number":"119168","commit":"b928addc472d9e74ea855c7617dd8f0b6e5d177c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119511/","number":"119511","commit":"b0093d1c0c78e0831ca4e2628f0706e0a189dd83"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119631/","number":"119631","commit":"104717da112829974f98351b2e3394dc39da972e"}]}
PIPELINE-->

majagrubic commented:
@kertal mind if I do this in a follow-up PR? seems like it's gonna be a decently-sized chunk of work

kertal commented:
> @kertal mind if I do this in a follow-up PR? seems like it's gonna be a decently-sized chunk of work

fine with that!

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/122246/)
* Commit: 104717da112829974f98351b2e3394dc39da972e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96762/104717da112829974f98351b2e3394dc39da972e)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ad628878b11f1252769a511b24a7c76a3dbe046a) | [after](https://github.com/elastic/kibana/commit/104717da112829974f98351b2e3394dc39da972e) | diff |
| --- | --- | --- | --- |
| `discover` | 402.7KB | 406.9KB | +4.2KB |

### History
* :green_heart: [Build #119631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119631/) succeeded 104717da112829974f98351b2e3394dc39da972e
* :green_heart: [Build #119511](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119511/) succeeded b0093d1c0c78e0831ca4e2628f0706e0a189dd83
* :green_heart: [Build #119168](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119168/) succeeded b928addc472d9e74ea855c7617dd8f0b6e5d177c
* :green_heart: [Build #118753](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/) succeeded 60151274baa6ef0017a0802a4e33a5ed0688c679
* :broken_heart: [Build #118703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/) failed b4a3003ba8b05d5328152b3962b0b1940d71e67f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118703/","number":"118703","commit":"b4a3003ba8b05d5328152b3962b0b1940d71e67f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118753/","number":"118753","commit":"60151274baa6ef0017a0802a4e33a5ed0688c679"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119168/","number":"119168","commit":"b928addc472d9e74ea855c7617dd8f0b6e5d177c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119511/","number":"119511","commit":"b0093d1c0c78e0831ca4e2628f0706e0a189dd83"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119631/","number":"119631","commit":"104717da112829974f98351b2e3394dc39da972e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/122246/","number":"122246","commit":"104717da112829974f98351b2e3394dc39da972e"}]}
PIPELINE-->
